stream_alert/alert_processor/helpers.py

```diff
@@ -30,3 +31,81 @@ def elide_string_middle(text, max_length):
 
     half_len = (max_length - 5) / 2  # Length of text on either side.
     return '{} ... {}'.format(text[:half_len], text[-half_len:])
+
+
+def publish_alert(alert, output, descriptor):
+    """Presents the alert as a dict for output classes to send to their API integrations.
+
+    Args:
+        alert (Alert): The alert to be dispatched
+        output (OutputDispatcher): Instance of the output class dispatching this alert
+        descriptor (str): The descriptor for the output
+
+    Returns:
+        dict
+    """
+    publisher = _assemble_alert_publisher_for_output(
+        alert,
+        output,
+        descriptor
+    )
+    return publisher.publish(alert, {})
+
+
+def _assemble_alert_publisher_for_output(alert, output, descriptor):
+    """Gathers all requested publishers on the alert and returns them as a single Publisher
+
+    Args:
+        alert (Alert): The alert that is pulled from DynamoDB
+        output (OutputDispatcher|None): Instance of OutputDispatcher that is sending the alert
+        descriptor (str): The descriptor of the Output
+
+    Returns:
+        AlertPublisher
+    """
+
+    alert_publishers = alert.publishers
+    publisher_names = []
+    if isinstance(alert_publishers, basestring):
+        # Case 1: The publisher is a single string.
+        #   apply this single publisher to all outputs + descriptors
+        publisher_names.append(alert_publishers)
+    elif isinstance(alert_publishers, list):
+        # Case 2: The publisher is an array of strings.
+        #   apply all publishers to all outputs + descriptors
+        publisher_names += alert_publishers
+    elif isinstance(alert_publishers, dict):
+        # Case 3: The publisher is a dict mapping output strings -> strings or list of strings
+        #   apply only publishers under the correct output key. We look under 2 keys:
+        #   one key that applies publishers to all outputs for a specific output type, and
+        #   another key that applies publishers only to outputs of the type AND matching
+        #   descriptor.
+
+        # FIXME (derek.wang)
+        # this is here because currently the OutputDispatcher sits in a __init__.py module that
+        # performs on-demand loading of the other output classes. If you load this helper before
+        # the output classes are loaded, it creates a cyclical dependency
+        # helper.py -> output_base.py -> __init__ -> komand.py -> helper.py
+        # This is a temporary workaround
```

Comment:
we should address this :)

Reply:
The `AlertPublisherLoader` is a pattern that I implemented to prevent myself from making the same mistake. 

The publishers are loaded on-demand from the repository, rather than immediately when you import the base module. This gives up compile-time import because everything has to happen at runtime. Maybe we can apply this to outputs too. Thoughts?